src/data_api/events.se

```diff
@@ -1,346 +1,420 @@
+    return(eventId)
+
+def report(eventId, outcome, attorepStake):
+    self.migrateEventThroughAllForks(eventId)
+    self.tryFinalizeEvent(eventId)
+    self.assertEventIsNotFinalized()
+
+    # self.assertInReportingPhaseOfCurrentWindow()
+    self.assertEventIsNotDoneWithReporters()
+    self.assertReporterHasRegistrationToken(msg.sender)
+
+    self.privateBuyToken(eventId, outcome, msg.sender, attorepStake)
+    self.privateAdjustTentativeOutcome(eventId, outcome)
+    self.controller.lookup('reporting').noteUserReport(branch, eventId)
     return(1)
 
-def getResolveBondPoster(event):
-    return(self.Events[event].resolveBondPoster)
-
-def getChallenged(event):
-    return(self.Events[event].challenged)
-
-def setChallenged(event):
-    self.controller.assertIsWhitelisted(msg.sender)
-    self.Events[event].challenged = 1
+def redeemTokens(eventId, branch, outcome):
+    self.migrateEventThroughAllForks(eventId)
+    if (self.privateTryRedeemTokensForNonForkingEventOnForkedBranch(eventId, branch, outcome, msg.sender)):
+        return(1)
+    if (self.privateTryRedeemTokensOnForkingEvent(eventId, branch, outcome, msg.sender)):
+        return(1)
+    self.privateRedeemCorrectOutcomeTokens(eventId, branch, outcome, msg.sender)
     return(1)
 
-# @return fxp
-def getFirstPreliminaryOutcome(event):
-    return(self.Events[event].firstPreliminaryOutcome)
-
-def setFirstPreliminaryOutcome(event, outcome):
-    self.controller.assertIsWhitelisted(msg.sender)
-    self.Events[event].firstPreliminaryOutcome = outcome
-    return(1)
+def automatedReport(eventId, outcome):
+    # intentionally does not migrate the event because automated reporting may result in not needing to
+    if (self.events[eventId].automatedReporterAddress == 0):
+        ~invalid()
+    if (msg.sender != self.events[eventId].automatedReporterAddress):
+        ~invalid()
+    if (self.events[eventId].automatedReporterOutcome != 0):
+        ~invalid()
+    if (block.timestamp < self.events[eventId].endTime):
+        ~invalid()
+    if (block.timestamp > self.getAutomatedReportDueTimestamp(eventId)):
+        ~invalid()
+    self.events[eventId].automatedReporterOutcome = outcome
 
-def getReportersPaidSoFar(event):
-    return(self.Events[event].reportersPaidSoFarForEvent)
+def disputeAutomatedReport(eventId):
+    self.migrateEventThroughAllForks(eventId)
+    if (self.isEventFinalized(eventId)):
+        ~invalid()
+    if (self.events[eventId].automatedReporterOutcome == 0):
+        ~invalid()
+    if (self.events[eventId].automatedReporterDisputeBondHolder != 0):
+        ~invalid()
+    self.assertEventIsInAutomatedReportDisputePhase(eventId)
+    # self.controller.lookup('reputationToken').transferFrom(msg.sender, self, self.automatedReporterDisputeBondAmount)
+    self.events[eventId].automatedReporterDisputeBondHolder = msg.sender
+    self.events[eventId].automatedReporterOutcome = self.events[eventId].tentativeOutcome
```

Comment:
This will change a bit in @adrake33's branch re automated reporting, but for limited/all reporting the tentative outcome is whatever has been reported at the time of the dispute.